src/instruments/yokogawa/yokogawa6370.py

```diff
@@ -60,19 +57,36 @@ def __init__(self, parent, idx):
 
         # METHODS #
 
-        def data(self, bin_format=True):
-            cmd = f":TRAC:Y? {self._name}"
+        def _data(self, axis, limits=None, bin_format=True):
+            """Get data of `axis`."""
+            if limits is None:
+                cmd = f":TRAC:{axis}? {self._name}"
+            elif isinstance(limits, (tuple, list)) and len(limits) >= 2:
+                cmd = f":TRAC:{axis}? {self._name},{limits[0]+1},{limits[1]+1}"
+            else:
+                raise AssertionError(
+                    "limits has to be an list or tuple with at least two members"
+                )
             self._parent.sendcmd(cmd)
             data = self._parent.binblockread(data_width=8, fmt="<d")
             self._parent._file.read_raw(1)  # pylint: disable=protected-access
             return data
 
-        def wavelength(self, bin_format=True):
-            cmd = f":TRAC:X? {self._name}"
-            self._parent.sendcmd(cmd)
-            data = self._parent.binblockread(data_width=8, fmt="<d")
-            self._parent._file.read_raw(1)  # pylint: disable=protected-access
-            return data
+        def data(self, limits=None, bin_format=True):
+            """
+            Return the trace's level data.
+
+            :param limits: Range of samples to transfer. (0 to 50000)
+            """
```

Comment:
Can you add an example or two here? Maybe one with the `limits` argument used, since it has to have `>=2` entires? From this description, I'm not sure what to set there..., however, this might be due to my ignorance of the instrument.

Reply:
You need to give a lower and upper limit, so it have to be 2, actually.

Reply:
I improved the documentation and changed the test to "equal".